### PR TITLE
Applied latest changes of ApiManager

### PR DESCRIPTION
**Pull request by mangchiandjjoe**
- ServiceConfigRollout in  server_config.proto
- Check service config loading using ApiManager.LoadServiceRollouts()
- Added t test for service configs update using rollouts

### CODE REVIEW

File: src/nginx/t/config_rollouts_managed.t [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+is( $r->{uri}, '/v1/services/endpoints-test.cloudendpointsapis.com/configs/2016-08-25r3',
+  'Service config uri' );
+is( $r->{headers}->{host}, "127.0.0.1:${ServiceManagementPort}", 'Host header was set' );
+
+# validate rests of request has no change
+my $check_error_rest = 0;
+while($r = shift @servicemanagement_requests) {
+  if ($r->{verb} ne 'GET' ||
+    $r->{uri} ne '/v1/services/endpoints-test.cloudendpointsapis.com/rollouts?filter=status=SUCCESS' ||
+    $r->{headers}->{host} ne "127.0.0.1:${ServiceManagementPort}" ) {
```
Comment (qiwzhang) [inline src/nginx/t/config_rollouts_managed.t]: line 206 is not needed.  servicemanagement_requests is read from servicemanagement.log file which was wrotten by that http server.  Its "host" has to be that port.

File: src/nginx/t/config_rollouts_managed.t [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+
+# Backend requests verification, before and after config rollouts
+my @requests = ApiManager::read_http_stream( $t, 'bookstore.log' );
+is( scalar @requests, 2, 'Backend received one request' );
+
+$r = shift @requests;
+
+is( $r->{verb}, 'GET', 'Backend request was a get' );
+is( $r->{uri}, '/shelves?key=this-is-an-api-key', 'Backend uri was /shelves' );
+is( $r->{headers}->{host}, "127.0.0.1:${BackendPort}", 'Host header was set' );
```
Comment (qiwzhang) [inline src/nginx/t/config_rollouts_managed.t]: line 220 is not needed.

File: src/nginx/t/config_rollouts_managed.t [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+
+sub servicecontrol {
+  my ( $t, $port, $file, $done) = @_;
+  my $server = HttpServer->new( $port, $t->testdir() . '/' . $file )
+    or die "Can't create test server socket: $!\n";
+  local $SIG{PIPE} = 'IGNORE';
+  
+  my $index = 0;
+
+  $server->on( 'POST',
```
Comment (mangchiandjjoe) [inline src/nginx/t/config_rollouts_managed.t]: Updated

File: src/nginx/t/config_rollouts_managed.t [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+
+################################################################################
+
+sub bookstore {
+  my ( $t, $port, $file) = @_;
+  my $server = HttpServer->new( $port, $t->testdir() . '/' . $file )
+    or die "Can't create test server socket: $!\n";
+  local $SIG{PIPE} = 'IGNORE';
+
+  $server->on( 'GET', '/shelves?key=this-is-an-api-key', <<'EOF');
```
Comment (mangchiandjjoe) [inline src/nginx/t/config_rollouts_managed.t]: Updated

File: src/nginx/t/config_rollouts_managed.t (hunk 2 above)
Comment (mangchiandjjoe) [inline src/nginx/t/config_rollouts_managed.t]: Removed

File: src/nginx/t/config_rollouts_managed.t (hunk 1 above)
Comment (mangchiandjjoe) [inline src/nginx/t/config_rollouts_managed.t]: Removed

File: src/nginx/config.cc [hunk 5]
```diff
@@ -585,10 +587,8 @@ ngx_int_t ngx_esp_build_server_config(ngx_conf_t *cf, ngx_esp_loc_conf_t *lc,
       return NGX_ERROR;
     }
 
-    auto service_config = init_service_configs->Add();
-    service_config->set_service_config_file_full_path(
-        ngx_str_to_std(file_name));
-    service_config->set_traffic_percentage(100.0);
+    (*traffic_percentages)[ngx_str_to_std(file_name)] = 100.0;
+    service_config_rollout->set_rollout_id("");
```
Comment (mangchiandjjoe) [inline src/nginx/config.cc]: Removed

**Pull request merged**